src/raster/gcp.rs

```diff
@@ -20,11 +123,83 @@ impl Dataset {
 
         unsafe { SpatialRef::from_c_obj(c_ptr) }.ok()
     }
+
+    /// Fetch GCPs.
+    ///
+    /// See: [`GDALDataset::GetGCPs`](https://gdal.org/api/gdaldataset_cpp.html#_CPPv4N11GDALDataset7GetGCPsEv)
+    pub fn gcps(&self) -> &[GcpRef] {
+        let len = unsafe { gdal_sys::GDALGetGCPCount(self.c_dataset()) };
+        let data = unsafe { gdal_sys::GDALGetGCPs(self.c_dataset()) };
+        unsafe { std::slice::from_raw_parts(data as *const GcpRef, len as usize) }
+    }
+
+    /// Assign GCPs.
+    ///
+    /// This method assigns the passed set of GCPs to this dataset, as well as
+    /// setting their coordinate system.
+    ///
+    /// See: [`GDALDataset::SetGCPs(int, const GDAL_GCP *, const OGRSpatialReference *)`](https://gdal.org/api/gdaldataset_cpp.html#_CPPv4N11GDALDataset7SetGCPsEiPK8GDAL_GCPPK19OGRSpatialReference)
+    pub fn set_gcps(&self, gcps: Vec<Gcp>, spatial_ref: &SpatialRef) -> Result<()> {
```

Comment:
Devil's advocate question: Any arguments for *not* consuming `gcps`? I've not learned best practice here; when to choose from:

* `gcps: Vec<Gcp>`
* `gcps: &Vec<Gcp>`
* `gcps: &[Gcp]`

The last one seems most flexible (since we end up copying the data anyway), but as I said, I'm still learning API best practices.

Reply:
String to CString conversion should reuse the allocation of there's space for an extra byte.

Whether that's a good enough reason or not, that's arguable.